source/ocsp-support/ocsp-support.rst

```diff
@@ -134,6 +134,41 @@ Drivers SHOULD validate OCSP Responses in the manner specified in `RFC
 6960: 3.2 <https://tools.ietf.org/html/rfc6960#section-3.2>`__ to the
 extent that their TLS library allows.
 
+Suggested OCSP Caching Behavior
+-------------------------------
+Drivers with sufficient control over their TLS library's OCSP
+behavior SHOULD implement an OCSP cache. The key for this cache
+SHOULD be the certificate identifier (CertID) of the OCSP request
+as specified in `RFC 6960: 4.1.1
+<https://tools.ietf.org/html/rfc6960#section-4.1.1>`__.
+For convenience, the relevant section has been duplicated below\:
+
+.. code:: ASN.1
+
+   CertID          ::=     SEQUENCE {
+       hashAlgorithm       AlgorithmIdentifier,
+       issuerNameHash      OCTET STRING, -- Hash of issuer's DN
+       issuerKeyHash       OCTET STRING, -- Hash of issuer's public key
+       serialNumber        CertificateSerialNumber }
+
+If a driver would accept a conclusive OCSP response (stapled or
+non-stapled), the driver SHOULD cache that response. We define a
+conclusive OCSP response as an OCSP response that indicates that a
+certificate is either valid or revoked. Thus, an unknown certificate
+status SHOULD NOT be considered conclusive, and the corresponding OCSP
+response SHOULD NOT be cached.
```

Comment:
I think we can use "MUST NOT" for the two "SHOULD NOT" instances in this sentence. We're saying that drivers SHOULD cache results, but I think we can be more rigid when it comes to instructing drivers about whether or not to cache.

Alternatively, are you using SHOULD here because some drivers may have no control over the ability to (a) use a cache and (b) control what added to that cache? If so, then I suppose SHOULD makes sense.

Reply:
Using MUST would be nice. The limiting factor, as you correctly suspected, is in fact due to potential limitations in drivers' control over their OCSP caching behavior. Dotnet for example uses a cache, but has zero control over what happens with that cache.